libvast/vast/system/collector.hpp

```diff
@@ -0,0 +1,49 @@
+/******************************************************************************
+ *                    _   _____   __________                                  *
+ *                   | | / / _ | / __/_  __/     Visibility                   *
+ *                   | |/ / __ |_\ \  / /          Across                     *
+ *                   |___/_/ |_/___/ /_/       Space and Time                 *
+ *                                                                            *
+ * This file is part of VAST. It is subject to the license terms in the       *
+ * LICENSE file found in the top-level directory of this distribution and at  *
+ * http://vast.io/license. No part of VAST, including this file, may be       *
+ * copied, modified, propagated, or distributed except according to the terms *
+ * contained in the LICENSE file.                                             *
+ ******************************************************************************/
+
+#pragma once
+
+#include <cstdint>
+#include <string>
+
+#include <caf/detail/unordered_flat_map.hpp>
+#include <caf/fwd.hpp>
+
+#include "vast/ids.hpp"
+#include "vast/uuid.hpp"
+
+namespace vast::system {
+
+/// Maps partition IDs to EVALUATOR actors (1 per layout in the partition).
+using query_map = caf::detail::unordered_flat_map<uuid,
+                                                  std::vector<caf::actor>>;
```

Comment:
Why is this type alias in the header file? It's unused here.

Reply:
It's used by all consumers of this header (test/system/collector.cpp, src/system/collector.cpp, and src/system/index.cpp), because the EVALUATOR reacts to messages of type `(expression, query_map, actor)`.